lib/utils/CSVDatabase.h

```diff
@@ -271,6 +288,18 @@ class CSVDatabase : public Database
         std::vector<std::string> &data, 
         bool append);
 
+    /**
+     * @brief Count the number of lines of CSV database file.
+     *
+     * @pre !key.empty()
+     *
+     * @param[in] key The key associated with the vector of strings to be
+     *                read.
+     */
+    int
+    getLineCount(
+        const std::string& key);
```

Comment:
I see that other functions also use the argument name `key`, but isn't it just the filename? Would it be clearer to replace `key` with `filename`?

Reply:
I actually had the same question before. `filename` is more clear. I will amend the variable name in both `CSVDatabase` and `HDFDatabase`.

Reply:
I see. I will leave HDFDatabase as is. Just to note that the name of the argument `file_name` for the member functions in `CSVDatabase` will be different from `key` in `Database`.